File: UNI-APP/hksy_uni_3.0/pages/store/api/storeapi.js
import networking from "../../../public/networking.js"

// 获取商品信息
function getProductDetail(productId, completion) {
	networking.functionRequest("/getProductDetail", null, {
		"productId": productId
	}, null, completion)
}

// 商品列表(分页) 
function getProductList(page, limit, completion) {
	networking.functionRequest("/getProductList", null, {
		"size": limit,
		"current": page
	}, null, completion)
}

// 购买产品
function buyProduct(userId, productId, paymentCode, completion) {
	networking.functionRequest("/buyProduct", null, {
		"userId": userId,
		"productId": productId,
		"paymentCode": paymentCode
	}, null, completion, false, 1500)
}

module.exports = {
	getProductDetail: getProductDetail,
	getProductList: getProductList,
	buyProduct: buyProduct
}

File: Flutter/hksy_flutter/lib/pages/store/api/storeapi.dart
import 'package:hksy_flutter/public/networking.dart';

class StoreApi {
  static getProductList(page, limit, finish) {
    Networking.requestBase(
      "getProductList",
      finish,
      params: {
        "page": page,
        "limit": limit,
      },
    );
  }

  static getProductDetail(productId, finish) {
    Networking.requestBase(
      "getProductDetail",
      finish,
      params: {
        "productId": productId,
      },
    );
  }
}
